day04/lib/main.rb

```diff
@@ -0,0 +1,114 @@
+# => TODO
+# => warehouse (how many product we have avaible)
+# => total price in basket + tax
+
+class Main
+  require_relative "./product"
+  require_relative "./warehouse"
+  require_relative "./basket"
+
+  attr_reader :action
+  def initialize
+    get_basket
+    get_list
+  end
+
+  def get_list
+    @get_list ||= Warehouse.new.make_list
+  end
+
+  def get_basket
+    bas = Basket.new
+    @basket ||= bas.create_basket
```

Comment:
This is too much complication. The initialization that you have now in `create_basket` method should go to `initialize` in the `Basket` class.

Then you don't need the temporary variable. Just use `@basket = Basket.new`. I think that then it would be better to also set this in the initializer in Main with attr_reader